target/linux/ath79/image/generic.mk

```diff
@@ -421,6 +421,20 @@ define Device/netgear_wndr3x00
   DEVICE_PACKAGES := kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-usb-ledtrig-usbport kmod-leds-reset kmod-owl-loader
 endef
 
+define Device/netgear_ex7300
+  ATH_SOC := qca9558
+  IMAGE/default := append-kernel | pad-offset $$$$(BLOCKSIZE) 64 | netgear-rootfs | pad-rootfs
+  $(Device/netgear_ath79)
+  DEVICE_TITLE := NETGEAR EX6400 or EX7300
+  NETGEAR_KERNEL_MAGIC := 0x27051956
+  NETGEAR_BOARD_ID := EX7300series
+  NETGEAR_HW_ID := 29765104+16+0+128
+  IMAGE_SIZE := 15552k
+  SUPPORTED_DEVICES += ex7300
```

Comment:
You can drop this, it's being added automatically for you in https://github.com/openwrt/openwrt/blob/eb1887be93a765b4b86ecad14dd72a5a72d6870e/target/linux/ath79/image/Makefile#L63

Reply:
Uh-oh. This may break what I did in the other commit, as well as the existing "wndr3x00" entry.

Reply:
On second look, I'm not sure you are correct, and the other Netgear devices all have such a line.

Reply:
He is correct, while his explanation is misleading. The correct supported devices entry for your board is added by the mentioned code line, to ensure sysupgrade accepts the image.

The extension of the SUPPORTED_DEVICES is only required if the board used a different userspace boardname in the past and you want to upgrade from an image using the old name to this image (which is using the new name).

Since you add support for a new board, you don't need to take care of former used userspace boardnames.

Reply:
In that case, is what I did in the f55a944 commit incorrect?